### PR TITLE
addActualItemCountOnRequesetLevel

### PR DESCRIPTION
**Pull request by xinlian12**
**Changes included:**
Added following metrics:
`cosmos.client.req.rntbd.actualItemCount`
`cosmos.client.req.gw.actualItemCount`

image

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java [hunk 1]
```diff
@@ -1098,6 +1123,26 @@ private void recordGatewayStatistics(
                     }
                 }
 
+                if (actualItemCount > 0) {
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java:1126]: One thing I learned recently is that Prometheus will cause issues when you sometimes add a tag and sometimes not. So, if this tag is enabled (like if actualItemCount>1 we would publish it) we should always publish it - use 0 when necessary.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java [hunk 2]
```diff
@@ -985,6 +989,26 @@ private void recordStoreResponseStatistics(
                     requestCounter.increment();
                 }
 
+                if (actualItemCount > 0) {
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java:992]: condition should be removed - see comment below

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM except for one comment that please needs to be addressed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java]: removed the check

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java]: updated

**Review by kushagraThapar — APPROVED** [review]

**Comment by xinlian12** [comment]
Manual trigger the cosmos test pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3697388&view=results

failed tests passed locally: 
readFeedDocumentsAfterSplit

**Pull request merged**